### PR TITLE
Redesigning connection flow and home screen

### LINKED ISSUE #233: Refactor the connect to server user flow
**What's missing? Please describe your flow and where this feature will help.**
Based on #249 we need to refactor the connect to server user flow.

### LINKED ISSUE #258: Make the TUI fit available screen size
**Describe the bug**
When I edit a gate and the App is a 1/4 of the screen size there's an ugly overflow

**To Reproduce**
1. Launch app
2. Make the App a quarter of the screen wide or 375px wide
3. Long press a gate

**Screenshots**
IMG_1666

**Additional context**

### LINKED ISSUE #259: redesign home screen to a map
The new "home screen" will be a map of a terminal. The map will be drawn using tiles. The height grows based on the number of gates and the width is based on the screen's width and can either 1,2 and 4 tiles.

The change also includes changing the home icon to a very small map.

### PR DESCRIPTION
**Pull request by eyal-sasson**
Closes #233.
Edit: everything is done, also resolves #258 now.
Now also closes #259.

### CODE REVIEW

**Comment by eyal-sasson** [comment]
Also closes #259 now.

File: src/terminal7.js [hunk 1]
```diff
@@ -522,14 +469,16 @@ peer_name = "${peername}"\n`
...
         this.confEditor = null
-        this.pbVerify()
-
+        if (this.pb &&
+            ((this.pb.host != this.conf.net.peerbook) 
+             || (this.pb.peerName != this.conf.peerbook.peer_name)
+             || (this.pb.insecure != this.conf.peerbook.insecure)
+             || (this.pb.email != this.conf.peerbook.email))
+        )
+            this.pb.close()
```
Comment (daonb) [inline src/terminal7.js:478]: Please add `this.pb = null` after the close

File: tests/gate.test.js [hunk 2]
```diff
@@ -110,17 +112,17 @@ describe("gate", () => {
...
+        t.addGate({name:"foo", addr: "foo"})
+        valid = t.validateHostName("foo")
         expect(valid).equal("Name already taken")
     })
     it("can be connected", async () => {
         let g = t.addGate()
         expect(typeof g).toEqual("object")
         g.open(e)
-        g.tryWebexec = false
+        // g.tryWebexec = false
```
Comment (daonb) [inline tests/gate.test.js:125]: remove commented code

**Review by daonb — CHANGES_REQUESTED** [review]

File: src/gate.ts [hunk 3]
```diff
@@ -386,14 +508,14 @@ echo "${fp}" >> ~/.config/webexec/authorized_fingerprints
      */
     addWindow(name, createPane) {
         this.t7.log(`adding Window: ${name}`)
-        let id = this.windows.length
-        let w = new Window({name:name, gate: this, id: id})
+        const id = this.windows.length,
+			w = new Window({name:name, gate: this, id: id})
```
[the remark this answers is not in the document]
Comment (eyal-sasson) [reply src/gate.ts:512]: It is a const, there's a comma above

File: src/form.ts [hunk 4]
```diff
@@ -124,14 +127,17 @@ export class Form {
             t.writeln("  " + this.fields.map(f => `  ${f.prompt}`).join('\n  '))
             t.write(`\x1B[3G\x1B[${len}A`) // move cursor to first field
             t.write(`\x1B[1m  ${this.fields[current].prompt}\x1B[0m\x1B[3G`)
+			Form.disposeCurrent = () => {
```
Comment (daonb) [inline src/form.ts:130]: Please remove a duplication of 63:68

File: src/form.ts [hunk 5]
```diff
@@ -41,10 +41,10 @@ export class Form {
     i: number
     e: HTMLElement
     resolve: (value) => void
-    reject: (reason?) => void
     fields: Fields
     results: Results
     static activeForm = false
+	static disposeCurrent: () => void
```
Comment (daonb) [inline src/form.ts:47]: This is a "constant" function , better defined it the typescript way. Also, we need a better name, so please change to:

```typescript
// closes the active form 
static close() {
   ...
}
```

File: src/form.ts [hunk 6]
```diff
@@ -206,7 +215,7 @@ export class Form {
                                 }
                             })
                         }
-                        else {
+                        else if (key.length == 1) {
```
Comment (daonb) [inline src/form.ts:218]: What are we ignoring? please add a comment on the line before

File: src/gate.ts [hunk 7]
```diff
@@ -260,9 +260,8 @@ export class Gate {
                     Storage.set({key: "first_gate", value: "1"}) 
                 }
             })
-            if (this.onConnected)
-                this.onConnected()
-                this.onConnected = this.load
+            this.onConnected()
+            this.onConnected = this.load
```
Comment (daonb) [inline src/gate.ts:264]: Add a comment before:

```typescript
// first onConnected is special if it's a new gate but once connected, we're back to load

File: index.html [hunk 8]
```diff
@@ -18,6 +18,13 @@
 <body>
   <div id="terminal7">
       <div id="log" class="border hidden">
+		  <div id="terminal-container"></div>
+		  <nav>
```
Comment (daonb) [inline index.html:22]: This came out pretty ugly. please remove the X button

**Review by daonb — CHANGES_REQUESTED** [review]
Almost there...

**Pull request merged**